processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -900,6 +927,27 @@ public Rowboat apply(@Nullable Rowboat input)
         )
     );
 
+    if (segmentMetadata != null && !segmentMetadata.isEmpty()) {
+      File metadataFile = new File(v8OutDir, "metadata.drd");
+      FileOutputStream metadataFileOutputStream = null;
+      FileChannel metadataFilechannel = null;
+      try {
+        metadataFileOutputStream = new FileOutputStream(metadataFile);
+        metadataFilechannel = metadataFileOutputStream.getChannel();
+        metadataFilechannel.write(ByteBuffer.wrap(mapper.writeValueAsBytes(segmentMetadata)));
```

Comment:
Reading through the docks and peeking at the code, it is not immediately clear what happens if the file fails to write due to lack of disk space. Purely from reading the javadoc both emitting IOException or simply writing what it could and returning the number of bytes written could both be correct behavior.

Suggest checking the return value to make sure the totality of expected bytes were written.